src/scripts/home.js

```diff
@@ -304,22 +266,131 @@ function delete_profile(profile) {
   req_card.remove();
   // Remove profile from list and save list
   profile_list = profile_list.filter(curr_prof => curr_prof.id !== profile.id);
+  TAG_MAP.set(profile.tag, TAG_MAP.get(profile.tag) - 1);
+  if (TAG_MAP.get(profile.tag) === 0) {
+    TAG_MAP.delete(profile.tag)
+    create_tag_btn();
+  }
+  console.log(TAG_MAP)
   save_profile_to_storage();
 }
 
-/**
- * display all the cards with the same tag on main page
- *
- * 1. for each profile in profile_list, check if it has
- *    such a tag. if it has, add this profile object to a
- *    temp list.
- * 2. remove all card components in grid
- * 3. call add_profiles_to_doc(profiles) to display the
- *    profile we just added to the temp list
- *
- * @param {String} tag an tag
- */
-function sort_by_tag(tag) {}
+function search_tag(tag) {
+  const match_list = [];
+  profile_list.forEach((profile) => {
+    if (
+      profile.tag &&
+      !match_list.includes(profile.tag) &&
+      profile.tag.includes(tag)
+    ) {
+      match_list.push(profile.tag);
+    }
+  });
+  return match_list;
+}
+
+function create_tag_btn() {
```

Comment:
documentation pls

Reply:
this one im having trouble understanding rn, my brain is too fried rn since its late